### PR TITLE
add xk6-chaos summary

### PR DESCRIPTION
**Pull request by simonaronsson**

### CODE REVIEW

File: chaos.go [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 )
 
-const version = "v0.0.1"
+const version = "v0.0.2"
 
 func init() {
 	modules.Register("k6/x/chaos", &Chaos{
 		Version: version,
 	})
+	fmt.Println("Running simskij/xk6-chaos@$" + version)
```
Comment (simonaronsson) [inline chaos.go]:
```suggestion
	fmt.Println("Running k6io/xk6-chaos@$" + version)
```

File: internal/experiments/experiments.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"context"
+
+	"github.com/simskij/xk6-chaos/internal/experiments/podkillers"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+)
+
+var ChaosSummary string
```
Comment (simonaronsson) [inline internal/experiments/experiments.go]:
```suggestion
```

File: internal/experiments/experiments.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// XPodkillers serves as a constructor of the Podkillers js class
+func (*Experiments) XPodkillers(ctx *context.Context) (interface{}, error) {
+	rt := common.GetRuntime(*ctx)
+	p := podkillers.New()
+	return common.Bind(rt, p, ctx), nil
+}
+
+func (*Experiments) GenerateChaosSummary() string {
+	return ChaosSummary
+}
```
Comment (simonaronsson) [inline internal/experiments/experiments.go]:
```suggestion
```

File: internal/experiments/podkillers/podkillers.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+// KillRandomPod terminates a pod at random.
+func (p *Podkillers) KillRandomPod(namespace string) error {
+	p.GetStartingPods(namespace)
+	podName, err := p.pod.KillRandom(context.Background(), namespace)
+	p.AddResults(namespace, podName)
+	// fmt.Println(p.Timestamp() + "Random pod " + podName + " terminated.")
+	return err
+}
```
Comment (simonaronsson) [inline internal/experiments/podkillers/podkillers.go]:
```suggestion
// SetStartingPods saves the number of pods at the beginning of a test as the "before" state.
func (p *Podkillers) SetStartingPods(number int) {
	p.NumOfPodsBefore = number
}

// GetStartingPods returns and saves the number of pods at the beginning of the test for reporting purposes.
func (p *Podkillers) GetStartingPods(namespace string) int {
	var podsAlive, _ = p.pod.List(context.Background(), namespace)
	p.NumOfPodsBefore = len(podsAlive)
	return p.NumOfPodsBefore
}

// GetNumOfPods returns and saves the current number of pods.
func (p *Podkillers) GetNumOfPods(namespace string) int {
	time.Sleep(5 * time.Second)
	var podsAlive, _ = p.pod.List(context.Background(), namespace)
	p.NumOfPodsAfter = len(podsAlive)
	return p.NumOfPodsAfter
}

// KillPod terminates a k8s pod identified by name.
func (p *Podkillers) KillPod(namespace string, podName string) error {
	p.GetStartingPods(namespace)
	err := p.pod.KillByName(context.Background(), namespace, podName)
	p.AddResults(namespace, podName)
	return err
}

// KillPodLike terminates a k8s pod whose name contains string.
func (p *Podkillers) KillPodLike(namespace string, keyword string) error {
	p.GetStartingPods(namespace)
	podName, err := p.pod.KillByKeyword(context.Background(), namespace, keyword)
	p.AddResults(namespace, podName)
	return err
}

// KillRandomPod terminates a pod at random.
func (p *Podkillers) KillRandomPod(namespace string) error {
	p.GetStartingPods(namespace)
	podName, err := p.pod.KillRandom(context.Background(), namespace)
	p.AddResults(namespace, podName)
	return err
}
```

One of he best things about version control is that we don't need to keep commented out code anymore 🎉

File: internal/experiments/podkillers/podkillers.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	dt := time.Now()
+	tsMsg := dt.Format("2006-Jan-02 15:04:05")
+	return tsMsg
+}
+
+func (p *Podkillers) GenerateSummary() string {
+	sum := summary.GetSummary()
+	output := "\n\nxk6-chaos\n===\n\nPODKILLERS:\n"
+	for i, result := range sum.Results {
+		output += fmt.Sprintf(" Victim #%d: %s terminated at %s\n            Pods before: %d; Pods 5s after termination: %d\n", i, result.Victim, result.Timestamp, result.PodCount.Before, result.PodCount.After)
```
Comment (simonaronsson) [inline internal/experiments/podkillers/podkillers.go]:
```suggestion
		output += fmt.Sprintf(
		    " Victim #%d: %s terminated at %s\n            Pods before: %d; Pods 5s after termination: %d\n",
		    i, 
		    result.Victim, 
		    result.Timestamp, 
		    result.PodCount.Before, 
		    result.PodCount.After,
		)
```

File: internal/k8s/pods/pods.go [hunk 6]
```diff
@@ -40,10 +42,37 @@ func (pods *Pods) List(ctx context.Context, namespace string) ([]string, error)
...
+	// Randomly choose a pod from the given namespace
+	var podsList, err = pods.List(ctx, namespace)
+	var randomNum = rand.Intn(len(podsList) - 1)
+	var randomPod = podsList[randomNum]
+
+	// Kill the random pod by name
+	err = pods.KillByName(ctx, namespace, randomPod)
+	return randomPod, err
+}
+
```
Comment (simonaronsson) [inline internal/k8s/pods/pods.go]:
```suggestion
// KillByKeyword kills the first pod with a name that contains the specified keyword
func (pods *Pods) KillByKeyword(ctx context.Context, namespace string, podKeyword string) (string, error) {
	var podsList, err = pods.List(ctx, namespace)
	var podToCheck = ""
	for i := 0; i < len(podsList); i++ {
		podToCheck = podsList[i]
		if strings.Contains(podToCheck, podKeyword) {
			break
		}
	}
	var podName = podToCheck
	err = pods.KillByName(ctx, namespace, podName)
	return podName, err
}

// KillRandom kills a random pod within a namespace
func (pods *Pods) KillRandom(ctx context.Context, namespace string) (string, error) {
	var podsList, err = pods.List(ctx, namespace)
	var randomNum = rand.Intn(len(podsList) - 1)
	var randomPod = podsList[randomNum]

	err = pods.KillByName(ctx, namespace, randomPod)
	return randomPod, err
}

```

The intent is clearly expressed by the code itself, so no need for inline comments. 👍🏽

**Review by simonaronsson — COMMENTED** [review]
@nicolevanderhoeven, I'd suggest changing these things prior to merging 👍🏽

**Comment by nicolevanderhoeven** [comment]
Reopening after discussion with Simme!

**Pull request merged**